rmi/servercode/TransImpl/TransactionManager.java

```diff
@@ -201,7 +234,35 @@ public int getCounter() {
 		return this.transactionCounter;
 	}
 
-	public void createVault(ArrayList<RMHashtable> masterRecords) {
-		this.shadowVault = new HashVault(masterRecords);
+	public HashVault createNewVault() {
+		// initialize Vault
+		ArrayList<RMHashtable> default_tables = new ArrayList<RMHashtable>();
+		for (int i = 0; i < hashKey_index_start; i++) {
+				RMHashtable empty_table = new RMHashtable();
+				default_tables.add(empty_table);
+		}
+		System.out.println("Creating Empty Vault...");
+		HashVault emptyVault = new HashVault(default_tables);
+		return emptyVault;
+	}
+
+	// Want to store the vault to its stable file? Call this function
+  public void destroy(int transactionId) {
+    File shadowFile = new File("shadowVault_" + transactionId + ".ser");
+    if (shadowFile.exists()) {
+      shadowFile.delete();
+    }
+  }
+
+	public RMHashtable[] get_Masters() {
```

Comment:
When we fire up the MW, we need to recover our master records.
We will serialize those records in, afterwhich the MW will make a call to TM.get_Masters() in order to retrieve the hashtables for the 4 RM's.